### ISSUE TITLE
False positive of `unused_braces` when using methods inside of const-generic inside enum variant

### ISSUE THREAD

**Issue by datdenkikniet** [rust-lang/rust#145843]
### Code

```Rust
#![allow(dead_code)]
use core::num::NonZeroUsize;

const TEST: NonZeroUsize = NonZeroUsize::new(1).unwrap();

struct Test<const N: usize> {
    value: [u8; { TEST.get() }],
}
```

### Current output

```Shell
Compiling playground v0.0.1 (/playground)
warning: unnecessary braces around const expression
 --> src/lib.rs:7:17
  |
7 |     value: [u8; { TEST.get() }],
  |                 ^^          ^^
  |
  = note: `#[warn(unused_braces)]` on by default
help: remove these braces
  |
7 -     value: [u8; { TEST.get() }],
7 +     value: [u8; TEST.get()],
  |

warning: `playground` (lib) generated 1 warning (run `cargo fix --lib -p playground` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.76s
```

### Desired output

```Shell
Compiling playground v0.0.1 (/playground)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.59s
```

### Rationale and extra context

Applying the suggestion results in a compilation error as it is no longer valid.

I've tried my best to find duplicates, but have not managed to find any.

### Other cases

This is also broken in the latest nightly:

```Shell
rustc 1.91.0-nightly (809200ec9 2025-08-24)
binary: rustc
commit-hash: 809200ec956983fce4ae178b87dada69f01d0820
commit-date: 2025-08-24
host: x86_64-unknown-linux-gnu
release: 1.91.0-nightly
LLVM version: 21.1.0
```

### Rust Version

```Shell
rustc 1.89.0 (29483883e 2025-08-04)
binary: rustc
commit-hash: 29483883eed69d5fb4db01964cdf2af4d86e9cb2
commit-date: 2025-08-04
host: x86_64-unknown-linux-gnu
release: 1.89.0
LLVM version: 20.1.7
```

**Comment by datdenkikniet** [comment]
Ahh crap, yeah, this was entirely our own fault. An update to `postcard` fixed this problem. I just got myself twisted up in the fact that the warning showed up right in the struct definition, but it's nothing the compiler was doing.

Closing, and thank you for pointing out my mistake so quickly so I had to re-check this report @GrigorenkoPV :)
